### PR TITLE
Use allowed_mentions when sending user input

### PR DESCRIPTION
**Pull request by onerandomusername**
A few quick fixes to prevent abuse mentioning users.

image

#### Important Note:
The tag cog has been modified to now store content without cleaning mentions as `allowed_mentions` is utilized to show the tag contents.

### CODE REVIEW

File: cogs/remind.py [hunk 1]
```diff
@@ -139,7 +139,7 @@ async def on_reminder_complete(self, record):
 
 		try:
 			if channel is not None:
-				await channel.send(content=f'<@{user_id}>', embed=e)
+				await channel.send(content=f'<@{user_id}>', embed=e, allowed_mentions=discord.AllowedMentions(users=True))
```
Comment (Run1e) [inline cogs/remind.py:142]: Can probably be replaced with `allowed_mentions=discord.AllowedMentions(users=[discord.Object(user_id)])`

File: cogs/tags.py [hunk 2]
```diff
@@ -363,7 +363,8 @@ async def raw(self, ctx, *, tag_name: TagViewConverter()):
 		'''View raw contents of a tag. Useful when editing tags.'''
 
 		tag_name, record = tag_name
-		await ctx.send(discord.utils.escape_markdown(record.get('content')))
+		await ctx.send(discord.utils.escape_markdown(record.get('content')),
```
Comment (Run1e) [inline cogs/tags.py:366]: Use

f(
  a,
  b
)

instead of

f(a
  b)

File: cogs/tags.py [hunk 3]
```diff
@@ -462,7 +463,9 @@ async def transfer(self, ctx, tag_name: TagEditConverter(), *, new_owner: discor
 		res = await self.db.execute('UPDATE tag SET user_id=$1 WHERE id=$2', new_owner.id, record.get('id'))
 
 		if res == 'UPDATE 1':
-			await ctx.send('Tag \'{}\' transferred to \'{}\''.format(record.get('name'), new_owner.display_name))
+			await ctx.send('Tag \'{}\' transferred to \'{}\''.format(record.get('name'), new_owner.display_name),
```
Comment (Run1e) [inline cogs/tags.py:466]: Use

f(
a,
b
)

instead of

f(a
b)

File: cogs/tags.py [hunk 4]
```diff
@@ -344,7 +344,7 @@ def msg_check(message):
 			if content is None:
 				old_message = ctx.message
 				ctx.message = message
-				content = self.craft_tag_contents(ctx, await commands.clean_content().convert(ctx, message.content))
+				content = self.craft_tag_contents(ctx, message.content)
```
Comment (Run1e) [inline cogs/tags.py:347]: Nice catch

File: ace.py [hunk 5]
```diff
@@ -328,9 +328,15 @@ async def new(self, query, args, limit, timeout, return_status=False):
 	log.info('Creating postgres pool')
 	db = await asyncpg.create_pool(DB_BIND)
 
+	# create allowed mentions
+	log.debug('Creating allowed mentions')
```
Comment (Run1e) [inline ace.py:332]: Unneccesary logging

File: cogs/tags.py [hunk 6]
```diff
@@ -219,23 +219,23 @@ async def tag(self, ctx, *, tag_name: TagViewConverter = None):
...
+		await ctx.send(record.get('content'), allowed_mentions=discord.AllowedMentions.none())
 
 		await self.db.execute(
 			'UPDATE tag SET uses=$2, viewed_at=$3 WHERE id=$1',
 			record.get('id'), record.get('uses') + 1, datetime.utcnow()
 		)
 
 	@tag.command(aliases=['add', 'new'])
-	async def create(self, ctx, tag_name: tag_create_converter, *, content: commands.clean_content = None):
+	async def create(self, ctx, tag_name: tag_create_converter, *, content: str = None):
```
Comment (Run1e) [inline cogs/tags.py:230]: Why isn't content being cleaned anymore?

File: cogs/tags.py [hunk 7]
```diff
@@ -219,23 +219,23 @@ async def tag(self, ctx, *, tag_name: TagViewConverter = None):
...
 		'''Create a new tag.'''
 
 		await self.create_tag(ctx, tag_name, self.craft_tag_contents(ctx, content))
 		await ctx.send(f'Tag \'{tag_name}\' created.')
 
 	@tag.command()
 	@can_prompt()
-	async def edit(self, ctx, tag_name: TagEditConverter(), *, new_content: commands.clean_content = None):
+	async def edit(self, ctx, tag_name: TagEditConverter(), *, new_content: str = None):
```
Comment (Run1e) [inline cogs/tags.py:238]: Why isn't content being cleaned anymore?

**Review by Run1e — CHANGES_REQUESTED** [review]
Since the clean_content converted is being used a lot it could be statically instantiated at the top of the file and just referenced where it's needed, but that may be unnecessary.

File: cogs/tags.py (hunk 7 above)
Comment (onerandomusername) [reply cogs/tags.py:238]: Content no longer needs to be cleaned as by utilizing allowed mentions it won't actually ping any mentions in the tag.

This allows putting mentions for users and channels in tags as they won't ping anymore.
Allowed_mentions is niiice.

File: cogs/tags.py (hunk 6 above)
Comment (onerandomusername) [reply cogs/tags.py:230]: Content no longer needs to be cleaned as by utilizing allowed mentions it won't actually ping any mentions in the tag.

This allows putting mentions for users and channels in tags as they won't ping anymore.

**Pull request merged**